core/src/databases/database.rs

```diff
@@ -0,0 +1,61 @@
+use anyhow::{anyhow, Result};
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+
+use crate::{
+    databases::{
+        remote_databases::get_remote_database::get_remote_database,
+        table::{get_table_type_for_tables, Table, TableType},
+        table_schema::TableSchema,
+        transient_database::execute_query_on_transient_database,
+    },
+    stores::store::Store,
+};
+
+#[derive(Debug, Error)]
+pub enum QueryDatabaseError {
+    #[error("{0}")]
+    GenericError(#[from] anyhow::Error),
+    #[error("Too many result rows")]
+    TooManyResultRows,
+    #[error("Result is too large: {0}")]
+    ResultTooLarge(String),
+    #[error("Query execution error: {0}")]
+    ExecutionError(String),
+}
+
+#[derive(Debug, Deserialize, Serialize, Clone)]
+pub struct QueryResult {
+    pub value: serde_json::Value,
+}
+
+pub trait HasValue {
+    fn value(&self) -> &serde_json::Value;
+}
+
+impl HasValue for QueryResult {
+    fn value(&self) -> &serde_json::Value {
+        &self.value
+    }
+}
+
+pub async fn execute_query(
+    tables: &Vec<Table>,
+    query: &str,
+    store: Box<dyn Store + Sync + Send>,
+) -> Result<(Vec<QueryResult>, TableSchema), QueryDatabaseError> {
+    match get_table_type_for_tables(tables.iter().collect()) {
```

Comment:
FYI, i was looking at the implementation for this, it looks like we do not implement yet the "check that all remote db share the same secret_id", maybe add a todo ?

Reply:
We have it in this `get_table_type_for_tables` function. it errors if tables are not compatible

Reply:
Ah I see, I overlooked the Remote(String) (this is new to me but I like it)

Reply:
Yeah enums in rust are awesome. `get_table_type_for_tables` isn't that great as a name since it doesn't make it obvious it is checking that. Any idea how it could be better ?